src/server/native_controllers/index.js

```diff
@@ -1,11 +1,18 @@
 import path from 'path';
 import { get } from 'lodash';
 import { Observable } from 'rxjs';
+import Joi from 'joi';
 import { spawnNativeController } from './spawn_native_controller';
 import { findPluginSpecs } from '../../plugin_discovery';
 import { safeChildProcess } from '../../utils/child_process';
 import { NativeController } from './native_controller';
 
+const specSchema = Joi.object().keys({
+  pluginId: Joi.string().required(),
+  path: Joi.string().required(),
+  config: Joi.array().items(Joi.string())
+});
```

Comment:
Make sure to add `.default()` at the end here or `undefined` will validate successfully.

Reply:
I added a test for this, but the current schema validates properly against null items in the array.

Reply:
Yeah, it's only an issue with `undefined` specifically, which can't be encoded in JSON, so not an issue I guess